### PR TITLE
(Bug) Error requesting phone validation #2603

### PR DESCRIPTION
**Pull request by serdiukov-o-nordwhale**
# Description

- add silent updateContact error, just log it

About https://github.com/GoodDollar/GoodDAPP/issues/2554

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Review by sirpy — CHANGES_REQUESTED** [review]
1. restore failing tests. i'm fixing it
2. handle updateContact also in /verify/email

**Comment by serdiukov-o-nordwhale** [comment]
> * restore failing tests. i'm fixing it
done
btw i supposed they're not needed because method they're testing aren't exists anymore

> * handle updateContact also in /verify/email
done

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -506,38 +519,46 @@ const setup = (app: Router, verifier: VerificationAPI, gunPublic: StorageAPI, st
...
+      const hashedNewEmail = sha3(email)
       const currentEmail = user.email
 
-      log.debug('email verified', { user, body, verificationData, tempSavedMauticId, tempSavedEmail, currentEmail })
+      log.debug('email verified', {
+        user,
+        body,
+        verificationData,
+        tempSavedMauticId,
+        tempSavedEmail: email,
```
Comment (serdiukov-o-nordwhale) [inline src/server/verification/verificationAPI.js:531]:
```suggestion
        email,
        verificationData,
        tempSavedMauticId,        
```

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -506,38 +519,46 @@ const setup = (app: Router, verifier: VerificationAPI, gunPublic: StorageAPI, st
...
-              mauticId: tempSavedMauticId
-            }),
-            Mautic.updateContact(user.mauticId, { email: tempSavedEmail })
-          ])
+        if (runInEnv && mauticId) {
+          const updMauticPromise = Mautic.updateContact(mauticId, { email }).catch(e =>
+            log.error('Error updating Mautic contact', e.message, e, { mauticId, email })
+          )
+
+          await Promise.all([Mautic.deleteContact({ mauticId: tempSavedMauticId }), updMauticPromise])
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js:553]: lets catch also the deleteContact exception

**Review by sirpy — CHANGES_REQUESTED** [review]
1. restore failing tests. i'm fixing it
2. handle updateContact also in /verify/email

File: src/server/verification/verificationAPI.js (hunk 2 above)
Comment (serdiukov-o-nordwhale) [reply src/server/verification/verificationAPI.js]: done

**Comment by sirpy** [comment]
@serdiukov-o-nordwhale fix build

**Comment by serdiukov-o-nordwhale** [comment]
@sirpy I merged with the tests you've fixed in another PR and merged with master.
Now only two tests from fishManager are failing. Could you please look ?

**Pull request merged**